device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -487,39 +493,45 @@ public void close(IotHubConnectionStatusChangeReason reason, Throwable cause)
             throw new IllegalArgumentException("reason cannot be null");
         }
 
-        this.cancelPendingPackets();
+        this.isClosing = true;
```

Comment:
Set the flag outside of the synchronized block so that any currently running reconnection logic knows to give up when this flag is set to true. Then the rest of the ```close()``` code is in the synchronization block so that it waits for the reconnection logic to end before it starts

Reply:
Sounds like a good code comment